File: example-base-28/nodejs-utilities-playground-v2/src/dsa/strings/longest-substring-without-repeating.js
function longestSubstringLength(value) {
  const lastSeen = new Map();
  let left = 0, best = 0;
  for (let right = 0; right < value.length; right++) {
    const char = value[right];
    if (lastSeen.has(char) && lastSeen.get(char) >= left) left = lastSeen.get(char) + 1;
    lastSeen.set(char, right);
    best = Math.max(best, right - left + 1);
  }
  return best;
}
module.exports = { longestSubstringLength };

File: example-base-29/python-algorithms-playground-v2/algorithms/strings/longest_substring_without_repeating.py
def longest_unique_substring(s):
    left=best=0; seen={}
    for right,ch in enumerate(s):
        if ch in seen and seen[ch]>=left: left=seen[ch]+1
        seen[ch]=right; best=max(best,right-left+1)
    return best

if __name__ == "__main__": print(longest_unique_substring('abcabcbb'))
